src/perl/lib/wtsi_clarity/mq/me/charging/fluidigm.pm

```diff
@@ -0,0 +1,203 @@
+package wtsi_clarity::mq::me::charging::fluidigm;
+
+use Moose;
+use Readonly;
+
+use wtsi_clarity::dao::study_dao;
+use wtsi_clarity::util::uuid_generator qw/new_uuid/;
+
+with qw/ wtsi_clarity::mq::message_enhancer /;
+
+our $VERSION = '0.0';
+
+Readonly::Scalar my $EVENT_TYPE                 => q{billing};
+Readonly::Scalar my $CLARITY_PROJECT_ROLE_TYPE  => q{clarity_charge_project};
+Readonly::Scalar my $PROJECT_SUBJECT_TYPE       => q{clarity_project};
+Readonly::Scalar my $PRODUCT_TYPE               => q{Human QC 96:96};
+Readonly::Scalar my $PIPELINE                   => q{SM};
+Readonly::Scalar my $RESEARCHER_EMAIL           => q{res:researcher/email};
+
+## no critic(ValuesAndExpressions::RequireInterpolationOfMetachars)
+Readonly::Scalar my $PROJECT_LIMSID             => q{prj:project/@limsid};
+## use critic
+
+sub type {
+  return 'event'
+};
+
+sub _build__lims_ids {
+  # It has not got any lims_id.
+  return 1;
+};
+
+# @Override
+sub prepare_messages {
+  my $self = shift;
+
+  my %message = ();
+  $message{$self->type} = $self->_get_event_message();
+  $message{'lims'}      = $self->config->clarity_mq->{'id_lims'};
+
+  return [\%message];
+}
+
+sub _get_event_message {
+  my $self = shift;
+
+  my %event_message = ();
+  $event_message{'uuid'}            = $self->_get_uuid;
+  $event_message{'event_type'}      = $EVENT_TYPE;
+  $event_message{'occured_at'}      = $self->timestamp;
+  $event_message{'user_identifier'} = $self->_user_identifier;
+  $event_message{'subjects'}        = $self->_subjects;
+  $event_message{'metadata'}        = $self->_metadata;
+
+  return \%event_message;
+}
+
+sub _get_uuid {
+  my $self = shift;
+
+  return new_uuid();
```

Comment:
Shouldn't you save this somewhere so the uuid is the same each time this subject is used?